### PR TITLE
#73 - update readme

### PR DESCRIPTION
**Pull request by olenagerasimova**
Updated readme for #73.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#266](https://github.com/artipie/rpm-adapter/issues/266) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ea39274b-e530-4214-a7f3-60dadb6b1b0e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: README.md [hunk 1]
```diff
@@ -50,30 +53,37 @@ and instruct `Rpm` to update the meta info:
...
+
+```java
+Rpm rpm = new Rpm(storage, StandardNamingPolicy.SHA1, Digest.SHA256);
+```
+
+### Include filelists.xml metadata
+
+RPM repository may include filelists.xml metadata, this metadata is not required by all rpm package
+managers. By default, we generate this metadata file but this behaviour can be configured with the
+help of `Rpm` secondary constrictor.
```
Comment (olegmoz) [inline README.md:81]: @olenagerasimova `constrictor` is a typo

File: README.md [hunk 2]
```diff
@@ -41,7 +41,10 @@ This is the dependency you need:
 </dependency>
 ```
  
-Then, you implement `com.yegor256.rpm.Storage` interface.
+Then, you implement `com.artipie.asto.Storage` interface or use one the existing implementations,
```
Comment (olegmoz) [inline README.md:44]: @olenagerasimova I'd suggest to switch existing options and `you implement`. Obviously, own implementation of `Storage` is not a likely scenario

File: README.md [hunk 3]
```diff
@@ -50,30 +53,37 @@ and instruct `Rpm` to update the meta info:
 ```java
 import com.artipie.rpm.Rpm;
 Rpm rpm = new Rpm(storage);
-rpm.update("test/my.rpm");
+rpm.batchUpdate(new Key.From("rpm-repo"));
```
Comment (olegmoz) [inline README.md:56]: @olenagerasimova it's unclear what `rpm-repo` means in this example, what string should I pass here

File: README.md [hunk 4]
```diff
@@ -50,30 +53,37 @@ and instruct `Rpm` to update the meta info:
...
+`sha-256` algorithms for hashing. To change checksum calculation algorithm use secondary 
+constructor of `Rpm` and `Digest` enum to specify the algorithm:
+
+```java
+Rpm rpm = new Rpm(storage, StandardNamingPolicy.SHA1, Digest.SHA256);
+```
+
+### Include filelists.xml metadata
+
+RPM repository may include filelists.xml metadata, this metadata is not required by all rpm package
```
Comment (olegmoz) [inline README.md:79]: @olenagerasimova that would be better to quote `filelists.xml` here

**Comment by olenagerasimova** [comment]
@olegmoz thanks, updated, please, take a look

File: README.md [hunk 5]
```diff
@@ -41,39 +41,49 @@ This is the dependency you need:
 </dependency>
 ```
  
-Then, you implement `com.yegor256.rpm.Storage` interface.
+Then, use one of the existing `com.artipie.asto.Storage` implementations to create the `Storage`,
```
Comment (Vatavuk) [inline README.md:44]: @olenagerasimova It looks like something is missing in this sentence or a "." is missing.

File: README.md [hunk 6]
```diff
@@ -41,39 +41,49 @@ This is the dependency you need:
...
-as an argument. Finally, you put your artifacts to the storage
-and instruct `Rpm` to update the meta info:
+as an argument. Finally, you put your artifacts to the storage specifying repository key 
+(`rpm-repo` in our example) and instruct `Rpm` to update the meta info:
 
 ```java
 import com.artipie.rpm.Rpm;
-Rpm rpm = new Rpm(storage);
-rpm.update("test/my.rpm");
+final Rpm rpm = new Rpm(storage);
```
Comment (Vatavuk) [inline README.md:55]: @olenagerasimova maybe we can expand this example with instantiation of `FileStorage` jsut to make it more obvious.

File: README.md [hunk 7]
```diff
@@ -41,39 +41,49 @@ This is the dependency you need:
...
 for more technical details.
 
-### Naming policy
+### Naming policy and checksum computation algorithm
 
 RPM may use different names to store metadata files in the package,
-by default `NamingPolicy.DEFAULT` will be used, to change naming policy use
-secondary constructor of `Rpm` to configure it. For instance to add `SHA1` prefixes for metadata files
-use `new Rpm(storage, new NamingPolicy.Sha1Prefixed())`.
+by default we use `StandardNamingPolicy.PLAIN`, to change naming policy use
```
Comment (Vatavuk) [inline README.md:65]: @olenagerasimova This should be the end of sentence like `by default we use StandardNamingPolicy.PLAIN. To change naming policy use...`

**Comment by olenagerasimova** [comment]
@Vatavuk thanks, corrected, take a look please

**Comment by 0crat** [comment]
Job was finished in 25 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@54bbf007-af81-47f4-889b-85871a7a9b4d, thread: PQ-CT2E6TK9B
com.amazonaws.http.AmazonHttpClient$RequestExecutor[1359] com.amazonaws.ResetException: Failed to reset the request input stream;  If the request involves an input stream, the maximum stream buffer size can be configured via request.getRequestClientOptions().setReadLimit(int)
java.io.BufferedInputStream[170] java.io.IOException: Stream closed
```

[0.53.15](https://github.com/zerocracy/farm): CID: [f1588d81-c929-4276-aedf-7e14d83b0ed9](https://www.0crat.com/footprint/CT2E6TK9B/f1588d81-c929-4276-aedf-7e14d83b0ed9), Type: "Order was finished"

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**